src/lambda_function/dispatch/app.py

```diff
@@ -0,0 +1,141 @@
+"""
+Dispatch Lambda
+
+This Lambda polls the input-to-dispatch SQS queue and publishes to the dispatch SNS topic.
+Its purpose is to inspect each export and dispatch each file as a separate job in which
+the file will be decompressed and uploaded to S3.
+"""
+import json
+import logging
+import os
+import zipfile
+from urllib import parse
+
+import boto3
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def get_object_info(s3_event: dict) -> dict:
+    """
+    Derive object info from an S3 event.
+
+    Args:
+        s3_event (dict): An S3 event
+
+    Returns:
+        object_info (dict) The S3 object info
+    """
+    bucket_name = s3_event["s3"]["bucket"]["name"]
+    object_key = parse.unquote(s3_event["s3"]["object"]["key"])
+    object_info = {
+        "Bucket": bucket_name,
+        "Key": object_key,
+    }
+    return object_info
+
+def get_archive_contents(archive_path: str, bucket: str, key: str) -> list[dict]:
+    """
+    Inspect a ZIP archive for its file contents.
+
+    Args:
+        archive_path (str): The path of the ZIP archive to inspect.
+        bucket (str): The S3 bucket where the ZIP archive originates from.
+        key (str): The S3 key where the ZIP archive originates from.
+
+    Returns:
+        archive_contents (list) A list of dictionaries. Each dictionary contains
+            the keys:
+
+            * Bucket - The name of the S3 bucket
+            * Key - The S3 key
+            * Path - The path within the archive which identifies this file
+            * FileSize - The uncompressed size in bytes of this file
+    """
+    archive_contents = []
+    with zipfile.ZipFile(archive_path, "r") as archive:
+        for path in archive.infolist():
+            if (
+                "/" not in path.filename # necessary for pilot data only
+                and "Manifest" not in path.filename
+                and path.file_size > 0
+            ):
+                file_info = {
+                        "Bucket": bucket,
+                        "Key": key,
+                        "Path": path.filename,
+                        "FileSize": path.file_size
```

Comment:
Curious what the info `FileSize` will be used for later on? Automating the memory needed for each file?

Reply:
I don't have a specific use case in mind for it, but I think it's useful metadata to include. A Lambda might want to know if it's been tasked with decompressing a massive JSON file.